File: atciss/app/views/vatsim.py
from dataclasses import dataclass
from datetime import datetime

from pydantic import BaseModel
from vatsim.types import Pilot


@dataclass
class Traffic:
    callsign: str
    departure_icao: str
    departure: str
    destination_icao: str
    destination: str
    ac_type: str
    eta: datetime | None
    groundspeed: int

    def __lt__(self, other: "Traffic"):
        return self.eta < other.eta

    def __le__(self, other):
        return self.eta <= other.eta

    @classmethod
    def from_pilot(
        cls,
        pilot: Pilot,
        eta: datetime | None,
        dep: str | None,
        arr: str | None,
    ) -> "Traffic":
        return Traffic(
            callsign=pilot.callsign,
            departure_icao=pilot.flight_plan.departure,
            departure=dep or pilot.flight_plan.departure,
            destination_icao=pilot.flight_plan.arrival,
            destination=arr or pilot.flight_plan.arrival,
            ac_type=pilot.flight_plan.aircraft_short,
            groundspeed=pilot.groundspeed,
            eta=eta,
        )


class AerodromeTraffic(BaseModel):
    aerodrome: str
    arrivals: list[Traffic]
    departures: list[Traffic]

File: atciss-frontend/src/types/vatsim.ts
export interface Atis {
  cid: string
  name: string
  callsign: string
  frequency: string
  atis_code: string
  text_atis: string
  runways_in_use: string[]
}

export interface Traffic {
  callsign: string
  departure_icao: string
  departure: string
  destination_icao: string
  destination: string
  ac_type: string
  groundspeed: number
  eta?: string
}

export interface AerodromeTraffic {
  aerodrome: string
  arrivals: Traffic[]
  departures: Traffic[]
}

export interface Controller {
  cid: string
  name: string
  callsign: string
  frequency: string
  atis_code: string
  facility: number
  rating: number
  server: string
  visual_range: number
  logon_time: string
  last_updated: string
  text_atis: string
}
